src/ucp/tag/offload.c

```diff
@@ -78,59 +78,100 @@ void ucp_tag_offload_completed(uct_tag_context_t *self, uct_tag_t stag,
     ucp_request_complete_tag_recv(req, status);
 }
 
-static size_t ucp_tag_offload_fetch_rkey(ucp_ep_t *ep, ucp_sw_rndv_hdr_t *sw_hdr,
-                                         ucp_rndv_rts_hdr_t *rts)
+static UCS_F_ALWAYS_INLINE size_t
+ucp_tag_offload_rkey_size(ucp_context_t *ctx)
 {
-    uint64_t *addr;
-    size_t rkey_size;
+    ucp_worker_iface_t *iface;
+    ucp_md_index_t mdi;
+
+    iface = ucs_queue_head_elem_non_empty(&ctx->tm.offload.ifaces,
```

Comment:
maybe add check for non empty queue with assertion here and below? BTW, why it's always head?

Reply:
Currently, TM offload is enabled if we have just one TM capable interface only

Reply:
Ok, this assertion could help when we have it extended